BEPs/BEP-319.md

```diff
@@ -0,0 +1,91 @@
+
+## 2. Abstract 
+The proposal focuses on three key optimizations for the incentive mechanism of the Fast Finality feature:
+1. Making Fast Finality rewards governable while maintaining the basic block transaction fee allocation ratio.
+2. Ensuring a more balanced distribution of Fast Finality rewards between epochs.
+3. Extending the submission deadline for malicious voting evidence.
+
+## 3. Motivation 
+The introduction of the Fast Finality feature in BSC significantly reduces the time required for transaction final confirmation. This feature represents a substantial advantage for BSC compared to other EVM-compatible chains. The purpose of this BEP is to solidify and enhance this advantage.
+
+## 4. Specification
+### 4.1 Governable Rewards
+Currently, the reward is approximately 1/16 of the block transaction fees, leading to a lack of incentive for some validators to participate in voting, thereby reducing the stability expectation of the Fast Finality feature. This BEP makes Fast Finality rewards governable, laying the groundwork for future reward enhancements.
+#### 4.1.1 Existing Allocation Logic
+<div align="center">
+<img src=./assets/BEP-319/4.1-current-fee-distribution.png width=70% />
+</div>
+The existing allocation steps are as follows:
+
+1. Validators collect transaction fees from within the block.
+2. In the client, if the balance of the SystemRewardContract is less than 100 BNB, 1/16 of the collected transaction fees is allocated as rewards for relayers and Fast Finality voting. The remainder is deposited into the ValidatorContract.
+3. Within the ValidatorContract, 10% of the deposited amount is burned directly, and the remainder serves as potential earnings for validators and stakers.
+
+#### 4.1.2 Allocation Logic Moved to the Contract 
+<div align="center">
+<img src=./assets/BEP-319/4.2-new-fee-distribution.png width=90% />
+</div>
+Due to the deposition of the SystemRewardContract being in the client, requiring hard forks for each change, and for ease of governance, it is moved to the smart contract. The adjusted allocation logic is as follows:
+
+1. Validators collect transaction fees from within the block and deposit all into the ValidatorContract.
+2. Deposit finalityRewardRatio of transaction fees into the SystemRewardContract, where finalityRewardRatio is governable, with an initial value of 1/16.
+3. Within the ValidatorContract, 9.375% of the deposited amount is burned directly, and the remainder serves as potential earnings for validators and stakers.
+#### 4.1.3 Maintain Burn Ratio 
+The real-time burning mechanism is an essential part of the BNB deflation plan. In this BEP, the goal is to maintain it essentially unchanged. Before the implementation of this BEP, the proportion of burned transaction fees to block transaction fees is (1-1/16) * 10% = 9.375%. Therefore, after the implementation of this BEP, the burnRatio is modified to 9.375%.
```

Comment:
Before the BEP the burn ratio is 9.375%, and after the BEP it is 9.375% ? So what changed?

Reply:
as before
      burn ratio is `(1-1/16) * 10% `
now
      burn ratio is 1*9.375%
      
target is to keep the ratio unchange

Reply:
Before the BEP, the burn ratio will be: 9.375% -> 10%, depends on if the balance of systemreward is over 100BNB or not.
After the BEP, burn ratio will be fixed: 9.375%

Reply:
@buddh0 I think this part need to be updated.

Reply:
> Before the BEP the burn ratio is 9.375%, and after the BEP it is 9.375% ? So what changed?

updated

Reply:
> @buddh0 I think this part need to be updated.

accept